src/rtcModule/webrtc.cpp

```diff
@@ -775,30 +844,35 @@ void RtcModule::updatePeerAvState(Id chatid, Id callid, Id userid, uint32_t clie
     }
 }
 
-void RtcModule::removeCall(Id chatid, bool keepCallHandler)
+void RtcModule::removeCall(Id chatid, bool retry)
 {
     auto itCall = mCalls.find(chatid);
     Promise<void> pms = promise::_Void();
     if (itCall != mCalls.end())
     {
+        if (retry)
+        {
+            launchCallRetry(chatid, itCall->second->sentAv());
+        }
+
+        RTCM_LOG_DEBUG("Remove Call on state disconnected: %s", chatid.toString().c_str());
         pms = itCall->second->destroy(TermCode::kErrPeerOffline, false);
     }
 
     auto wptr = weakHandle();
-    pms.then([wptr, this, chatid, keepCallHandler]()
+    pms.then([wptr, this, chatid, retry]()
     {
         if (wptr.deleted())
             return;
 
         auto itHandler = mCallHandlers.find(chatid);
         if (itHandler != mCallHandlers.end())
         {
-            Chat &chat = mKarereClient.mChatdClient->chats(chatid);
-            if (keepCallHandler && itHandler->second->callParticipants() && chat.isGroup())
+            if (retry || itHandler->second->callParticipants())
             {
                 itHandler->second->removeAllParticipants();
             }
-            else
+            else if (mRetryCall.find(chatid) == mRetryCall.end())
```

Comment:
Better to add the new condition as an `assert()` in the `else` block.

Reply:
It's necessary keep the if condition to avoid destroy the call handler during reconnection

Reply:
In that case, since it doesn't seem obvious, maybe a comment would be helpful.